open-craft/opencraft
# [BB-2290] Add password recovery

This adds [django-rest-passwordreset](https://github.com/anexia-it/django-rest-passwordreset), which provides API for resetting password and adds the password recovery workflow to the new Ocim frontend.

**Testing instructions:**
1. Checkout this branch.
1. Run Ocim (inside vagrant with `make run.dev`).
1. Go to `frontend/` and build the new API client with `./scripts/build-api-client.sh`
1. Run the new frontend: `npm start`.
1. Create a new account in console.
1. Go to the `Login` page and click on `I forgot my password`.
1. Test the following:
   1. non-email value,
   1. non-existing value,
   1. existing value.
1. The last one from the previous point will output an email in the terminal that runs Ocim. Optionally you can copy the whole email and create HTML file from it to check that it looks correctly.
1. Copy the reset link from an email and paste into the browser. Try the following things on the password reset page:
   1. non-matching passwords,
   1. non-correct passwords,
   1. correct passwords.
1. After changing the password, you should be redirected to the login page with an appropriate `success` alert.
1. Use the copied link again and see that it's not valid anymore.

**Reviewers:**
